go/vt/workflow/node.go

```diff
@@ -329,6 +341,36 @@ func (m *NodeManager) broadcastUpdateLocked(u *Update) {
 	}
 }
 
+// updateNodeAndBroadcastLocked updates the contents of the Node saved inside node manager that
+// corresponds to the provided user node, and then broadcasts the contents to all watchers.
+// If updateChildren is false then contents of the node's children (as well as the list of
+// children) is not updated and not included into the broadcast.
+// Has to be called with the lock.
+func (m *NodeManager) updateNodeAndBroadcastLocked(userNode *Node, updateChildren bool) error {
+	savedNode, err := m.getNodeByPathLocked(userNode.Path)
+	if err != nil {
+		return err
+	}
+	if err := savedNode.deepCopyFrom(userNode, updateChildren); err != nil {
+		return err
+	}
+	var nodeToPush *Node
+	if updateChildren {
+		nodeToPush = savedNode
+	} else {
+		nodeToPush = NewNode()
```

Comment:
Instead of creating a new node, can we save the children, call broadcastUpdateLocked, and then restore the children? Would probably be less lines of code and more efficient overall.